src/ImageProcessor.Web.UnitTests/Extensions/DirectoryInfoExtensionsUnitTests.cs

```diff
@@ -0,0 +1,176 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="DirectoryInfoExtensionsUnitTests.cs" company="James South">
+//   Copyright (c) James South.
+//   Licensed under the Apache License, Version 2.0.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace ImageProcessor.Web.UnitTests.Extensions
+{
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+
+    using Web.Extensions;
+
+    using NUnit.Framework;
+
+    /// <summary>
+    /// The directory info extensions unit tests.
+    /// </summary>
+    public class DirectoryInfoExtensionsUnitTests
+    {
+        /// <summary>
+        /// The when safe enumerable directories.
+        /// </summary>
+        [TestFixture]
+        public class WhenSafeEnumerableDirectories
+        {
+            /// <summary>
+            /// The test directory root.
+            /// </summary>
+            private static readonly string TestDirectoryRoot = "DirectoryInfoExtensionsTests";
+
+            /// <summary>
+            /// The directory count.
+            /// </summary>
+            private readonly int directoryCount = 4;
+
+            /// <summary>
+            /// The directory list.
+            /// </summary>
+            private IEnumerable<string> directoryList;
+
+            /// <summary>
+            /// The setup directories.
+            /// </summary>
+            [SetUp]
+            public void SetupDirectories()
+            {
+                this.directoryList = Enumerable.Range(1, this.directoryCount).Select(i => $"{TestDirectoryRoot}/TestDirectory{i}");
```

Comment:
:smile: ....I thought about switching over but decided it was too risky for anyone collaborating. I'll need to check my AppVeyor settings as that didn't pick up on the issue.